File: new-skills/references/common/java/model/exception/WfException.java
/**
 * Alipay.com Inc.
 * Copyright (c) 2004-2026 All Rights Reserved.
 */
package {basePackage}.wf.model.exception;

/**
 * WorldFirst API 业务异常
 *
 * @author Qoder
 * @version WfException.java, v 0.1 2026-04-01
 */
public class WfException extends RuntimeException {

    private static final long serialVersionUID = 1L;

    /** 错误码 */
    private final WfErrorCode errorCode;

    /**
     * 构造函数
     *
     * @param errorCode 错误码枚举
     * @param message   错误描述
     */
    public WfException(WfErrorCode errorCode, String message) {
        super(message);
        this.errorCode = errorCode;
    }

    /**
     * 构造函数（带原始异常）
     *
     * @param errorCode 错误码枚举
     * @param message   错误描述
     * @param cause     原始异常
     */
    public WfException(WfErrorCode errorCode, String message, Throwable cause) {
        super(message, cause);
        this.errorCode = errorCode;
    }

    /**
     * Getter method for property <tt>errorCode</tt>.
     *
     * @return property value of errorCode
     */
    public WfErrorCode getErrorCode() {
        return errorCode;
    }

    /**
     * 是否可重试
     *
     * @return 可重试返回 true
     */
    public boolean isRetryable() {
        return errorCode != null && errorCode.isRetryable();
    }

    @Override
    public String toString() {
        return "WfException{errorCode=" + errorCode + ", message=" + getMessage() + '}';
    }
}

File: wf-api-integration/references/common/python/model/exception/wf_exception.py
# -*- coding: utf-8 -*-
"""
WorldFirst API 业务异常
"""

from {basePackage}.wf.model.exception.wf_error_code import WfErrorCode


class WfException(Exception):
    """WorldFirst API 业务异常

    Attributes:
        error_code: 错误码枚举
        message: 错误描述
    """

    def __init__(self, error_code: WfErrorCode, message: str, cause: Exception = None):
        """构造函数

        Args:
            error_code: 错误码枚举
            message: 错误描述
            cause: 原始异常（可选）
        """
        super().__init__(message)
        self.error_code = error_code
        self.cause = cause

    @property
    def retryable(self) -> bool:
        """是否可重试"""
        return self.error_code is not None and self.error_code.retryable

    def __repr__(self) -> str:
        return f"WfException(error_code={self.error_code}, message='{self}')"
